sql/src/main/java/io/crate/execution/dsl/projection/builder/SplitPoints.java

```diff
@@ -47,36 +44,25 @@
 public class SplitPoints {
 
     private final List<Symbol> toCollect;
-    private final List<Function> aggregatesOrTableFunctions;
+    private final List<Function> aggregates;
+    private final List<Function> tableFunctions;
 
-    @Nullable
-    private final FunctionInfo.Type functionsType;
 
-    SplitPoints(List<Symbol> toCollect,
-                List<Function> aggregatesOrTableFunctions,
-                @Nullable FunctionInfo.Type functionsType) {
+    SplitPoints(List<Symbol> toCollect, List<Function> aggregates, List<Function> tableFunctions) {
```

Comment:
Not sure if it's really worth it, but would it make sense to make SplitPoints accept a Collection<Symbol> as for the toCollect - at a high level view it seems we're for-eaching on this list when we use it

Reply:
I tried this but some components in the logical planner require lists (although not strictly necessary). I didn't want to change that everywhere in this PR, but I could follow up on it. Not sure if it's worth it